src/commands/Invoke.ts

```diff
@@ -1,30 +1,63 @@
-import fetch, { Request } from "node-fetch";
+import fetch, { BodyInit, Request, Response } from "node-fetch";
+import { ReadStream } from "fs";
+import queryString from "query-string";
+import { flatten } from "q-flat";
+import FormData from "form-data";
 
 import validateAndRefresh from "./RefreshIfNeeded";
 import retrieveImpersonatedCredentials from "./ImpersonateCredentials";
 
 import { Credentials, CredentialStorage } from "../oauth";
 import { ApiConfiguration } from "../ApiConfiguration";
-import queryString from 'query-string';
-import { flatten } from 'q-flat';
+
+export type InvokeBody = Record<string, unknown> | [] | ReadStream | Buffer;
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
-const createRequest = (url: string, config: ApiConfiguration, credentials: Credentials, payload?: Record<string, unknown> | []): Request => {
+const createRequest = (url: string, config: ApiConfiguration, credentials: Credentials, payload?: InvokeBody): Request => {
     const headers = {
         "Accept": "application/json",
         "Content-Type": "application/x-www-form-urlencoded",
         "Authorization": `Bearer ${credentials.accessToken}`
     };
+    let body: BodyInit | undefined;
+    if (url.endsWith('/trade-chat/image/upload') && payload) {
```

Comment:
As a quick hack that will do, but we need to have more flexible solution. In other words, this method should iterate values of payload and if any of them is of type, say, ArrayBuffer, then we would switch to `multipart/form-data` content-type.

Reply:
Yes I tried to do something like that but the amount of code + how complex it had become it doesn’t feel like paying for a just more simple implementation right now as we just have one endpoint that is `multipart/form-data`. If in future we had a new endpoint we can interact over this code again with new tests and improve the code accordingly. What do you think?

Reply:
Yep, for now will do, let's return to this topic later.